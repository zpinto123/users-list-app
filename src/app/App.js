import React, { Component } from "react";
import UsersList from "../features/usersList";
import "./App.css";

class App extends Component {
  render() {
    return (
      <div className="app">
        <UsersList />
      </div>
    );
  }
}

export default App;
