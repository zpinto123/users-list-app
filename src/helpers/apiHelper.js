import api from "../services/config";

const STATUS_OK = 200;

const fetchData = (url, method = "GET") =>
  fetch(url, { method, headers: setAuthHeaders() })
    .then(verifyResponse)
    .catch(handleErrors);

const verifyResponse = res => {
  if (!res || res.status !== STATUS_OK) {
    console.log(`error fetching - status: ${res.status}`);
    return res.status;
  }
  return res.json();
};

const handleErrors = error => console.error(error);

const setAuthHeaders = () => {
  const { username, password } = api;
  const headers = new Headers();
  headers.set(
    "Authorization",
    `Basic ${Buffer.from(`${username}:${password}`).toString("base64")}`
  );
  return headers;
};

export { fetchData };
