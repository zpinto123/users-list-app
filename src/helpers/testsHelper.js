import configureStore from "redux-mock-store";
import initialState from "../redux/modules/users/initialState";

const mockStore = configureStore();
const store = mockStore(initialState);

export { store };
