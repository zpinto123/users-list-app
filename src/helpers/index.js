import { fetchData } from "./apiHelper";

export { fetchData };
