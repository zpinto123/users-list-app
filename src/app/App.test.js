import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import { store } from "../helpers/testsHelper";
import App from "./App";

describe("<App />", () => {
  it("renders App correctly", () => {
    const renderer = new ShallowRenderer();
    const component = renderer.render(<App store={store} />);
    expect(component).toMatchSnapshot();
  });
});
