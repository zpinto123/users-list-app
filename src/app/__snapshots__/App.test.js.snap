// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders App correctly 1`] = `
<div
  className="app"
>
  <Connect(UsersList) />
</div>
`;
